### PR TITLE
Implementing TS-AppVersion

### PR DESCRIPTION
**Pull request by CFIALeronB**
- Installing and configuring TS-AppVersion.
- Ensuring seamless extraction of version information.
- Verifying compatibility with the TypeScript framework.
- Updated Footer version number display. Version number is pulled from TS-AppVersion.

#80

### CODE REVIEW

File: src/_versions.ts [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
+    gitCommitDate?: string;
+    gitTag?: string;
+};
 export const versions: TsAppVersion = {
-  version: "0.2.0",
-  name: "nachet-frontend",
-  versionDate: "2023-08-04T06:12:27.197Z",
-  gitCommitHash: "4c506f3",
-  versionLong: "0.1.0-4c506f3",
+    version: '0.1.0',
```
Comment (rngadam) [inline src/_versions.ts:12]: original had 0.2.0, now it's 0.1.0. 

 0.1.0 < 0.2.0

this should never happen.

The source of the version is from package.json. does not appear to be updated.

Workflow should check that version is increasing always.

https://github.com/saitho/ts-appversion/issues/74

**Review by rngadam — CHANGES_REQUESTED** [review]

File: src/_versions.ts (hunk 1 above)
Comment (CFIALeronB) [reply src/_versions.ts:12]: The _versions.ts file gets overwritten on npm start everytime. If the version number in the package.json is 0.2.0 then this file will be automatically updated to version: '0.2.0' during runtime.

Comment (rngadam) [reply src/_versions.ts:12]: yes, so it is required for the developer to update package.json version for each new version.

@ThomasCardin will soon enforce this with a github workflow check (making sure that the version of this PR is incremental)

**Review by rngadam — CHANGES_REQUESTED** [review]
update version in package.json

File: src/environments/environment.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import versions from "../_versions";
```
Comment (rngadam) [inline src/environments/environment.ts:1]: you have environment.prod, environment.staging. What is environment.ts? Is it the dev environment? so it should use versions.versionLong. How do we select environment based on target deployment (ie, how will we use the environment.staging for staging and environment.prod for production)?

**Review by rngadam — CHANGES_REQUESTED** [review]

File: src/environments/environment.ts (hunk 2 above)
Comment (CFIALeronB) [reply src/environments/environment.ts:1]: environment.ts is the default environment configuration. It's commonly used for development purposes. Settings in this file will be used when you're running the project locally or when you haven't specified a particular environment for deployment. You can select the environment based on the target deployment by setting the NODE_ENV environment variable to development, staging, or production, and then the application dynamically imports the corresponding environment.ts, environment.staging.ts, or environment.prod.ts configuration file. This is typically done in the build or run scripts of your package.json, or through environment variable settings in your CI/CD pipeline.

Comment (rngadam) [reply src/environments/environment.ts:1]: Thanks for the extensive explanation. This information should probably be in the README.md here, linking to a common frontend documentation in dev-rel-docs as we would want to make practices consistent between frontend components.

**Review by rngadam — CHANGES_REQUESTED** [review]

File: src/environments/environment.ts (hunk 2 above)
Comment (CFIALeronB) [reply src/environments/environment.ts:1]: @rngadam I've added further documentation to the TypeScript-AppVersion/APPVERSION-SETUP.md document, highlighting the structure and utilization of different environment configuration files. The update includes detailed guidelines on the purpose and usage of environment.ts, environment.staging.ts, and environment.prod.ts, 

I will update Nachet's README.md documentation and link the dev-rel-docs document once those PR changes are approved: https://github.com/ai-cfia/dev-rel-docs/pull/104

**Review by rngadam — APPROVED** [review]
you can link to the specific branch version of the environment configuration so we can move this PR forward even if the main doc is still ongoing.

**Review by RussellJimmies — APPROVED** [review]

**Pull request merged**